dynamiq/nodes/images/edit.py

```diff
@@ -0,0 +1,319 @@
+- Preserves original filenames in outputs
+
+Usage Strategy:
+- Provide clear, descriptive prompts for desired edits
+- Use masks to target specific areas for modification
+- Generate multiple variations to explore different results
+- Integrate seamlessly with agent workflows via file storage
+
+Parameter Guide:
+- prompt: Text description of desired edits (required)
+- image: Image file to edit, auto-injected from agent's file store
+- mask: Optional mask image to specify areas to edit
+- n: Number of edited versions to generate
+- size: Output dimensions (e.g., '1024x1024', '1792x1024')
+- response_format: 'url' or 'b64_json' output format
+
+Examples:
+- {"prompt": "Add a sunset background behind the subject", "image": <source_image>}
+- {"prompt": "Change the shirt color to blue", "image": <source_image>, "mask": <mask_image>}
+- {"prompt": "Make the image more vibrant and colorful", "n": 3, "image": <source_image>}"""
+    model: str = "gpt-image-1"
+    connection: OpenAIConnection | None = None
+    n: int | None = None
+    size: ImageSize | str = ImageSize.SIZE_1024x1024
+    response_format: ImageResponseFormat | str | None = Field(
+        default=None,
+        description="Response format (e.g., 'url', 'b64_json'). Only supported by some models. Will be dropped "
+        "if not supported.",
+    )
+    error_handling: ErrorHandling = Field(default_factory=lambda: ErrorHandling(timeout_seconds=600))
+    is_files_allowed: bool = True
+    input_schema: ClassVar[type[ImageEditInputSchema]] = ImageEditInputSchema
+
+    model_config = ConfigDict(extra="allow", arbitrary_types_allowed=True)
+
+    _image_edit: Callable = PrivateAttr()
+
+    def __init__(self, **kwargs):
+        """Initialize the ImageEdit node.
+
+        If neither client nor connection is provided, a new OpenAI connection is created.
+
+        Args:
+            **kwargs: Keyword arguments to initialize the node.
+        """
+        if kwargs.get("client") is None and kwargs.get("connection") is None:
+            kwargs["connection"] = OpenAIConnection()
+        super().__init__(**kwargs)
+
+        from litellm import image_edit
+
+        self._image_edit = image_edit
+
+    @property
+    def edit_params(self) -> dict:
+        """Get parameters for the image edit API call."""
+        params = self.connection.conn_params.copy() if self.connection else {}
+        if self.client:
+            params["client"] = self.client
+        if self.response_format is not None:
+            response_format_value = (
+                self.response_format.value
+                if isinstance(self.response_format, ImageResponseFormat)
+                else self.response_format
+            )
+            params["response_format"] = response_format_value
+
+        if hasattr(self, "__pydantic_extra__") and self.__pydantic_extra__:
+            import copy
+
+            extra = copy.deepcopy(self.__pydantic_extra__)
+            params.update(extra)
```

Comment:
Is it possible to use something like `extra_params: dict | None = None` in the node config instead of using internal Pydantic attributes?